dragon/console_prompt.rb

```diff
@@ -63,11 +64,44 @@ def move_cursor_left
         update_cursor_position_px
       end
 
+      # ? Works well enough, could be better
+      # TODO: Maybe possibly add skipping multiple word-breaking characters
+      # TODO: (requires re-writing a lot of the code most likely)
+      # ?? how can we put in regex? the code would be cleaner that way \ amir please fix :P \\ abolish consoles :>
+      def move_cursor_left_word
+        str = @current_input_str[0..[@cursor_position - 1, 0].sort[1]]
+        # ? Can be changed, it was just taken from my editor settings :>
+        @cursor_position = ("`~!@#$%^&*-=+()[]{}\|;:'\",.<>/?_ \t\n\0".chars.map { |char|
+          (val = str.rindex char) ? (val + 1 == @cursor_position ? val : val + 1) : 0
+        }.sort.reverse[0])
```

Comment:
Right now there's a lot of stuff going on here which might be hard to understand later if you try to improve it further after a few months

It would be easier to understand if you split it up in simpler steps like
- Find the first occurrence for each possible word boundary character
  ```rb
  word_boundary_positions = WORD_LIMITER_CHARS.map { |char| 
    # something with rindex
  }
  @cursor_position = word_boundary_positions.max
  ```

And same thing but reversed with `index` and `min` in `move_cursor_right_word`

Reply:
I am doing something similar in `move_cursor_right_word` as why I haven't done that here? Absolutely no idea

Reply:
Also we _do_ have `Array#max` and `Array#min` right? (my heart must've been in the right place but my brain was clearly out of the room when writing this code)

Reply:
Yes... max and min are available ;)

> my brain was clearly out of the room when writing this code
And well, I guess that's what a second pair of eyes is for :D